pkg/agent/storage/gcs.go

```diff
@@ -0,0 +1,117 @@
+package storage
+
+import (
+	gstorage "cloud.google.com/go/storage"
+	"context"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/googleapis/google-cloud-go-testing/storage/stiface"
+	"google.golang.org/api/iterator"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+type GCSProvider struct {
+	Client stiface.Client
+}
+
+func (p *GCSProvider) DownloadModel(modelDir string, modelName string, storageUri string) error {
+	gcsUri := strings.TrimPrefix(storageUri, string(GCS))
+	path := strings.Split(gcsUri, "/")
```

Comment:
this does not work for the case where only bucket is specified like `gs://bucket`, we fixed this in the S3Provider please checkout there.